### ISSUE TITLE
SingleResource uploading cant be supported

### ISSUE THREAD

**Issue by MJ1998** [google/android-fhir#2454]
**Describe the Issue**
With #2442, we are trying to solve sync issue involving referential integrity. Is this issue solvable in case of SingleResource uploading strategy ? Because in SingleResource the LocalChanges will be fetched out of order and one of them could still refer to a  Resource which would be synced later.

**Would you like to work on the issue?**

**Comment by aditya-07** [comment]
@MJ1998 Is this issue to find an alternate way to be able to use this strategy ?

**Comment by MJ1998** [comment]
@aditya-07 Yes Aditya. But I guess this can be deprioritized like you already have done!
